stdlib.ispc

```diff
@@ -3546,49 +3550,70 @@ __declspec(safe) static inline uniform float log(uniform float x_full) {
         z = x_full + y;
         return z + 0.693359375 * fe;
     } else if (__math_lib == __math_lib_ispc) {
+
+        // Precision: <10 ULP (for all input values).
```

Comment:
What about ULP for x=1.120036 (0x3f8f5d57)?

Reply:
The relative error formula gives 7.88694e-7 which looks like <10 ULP at first glance but as we discussed by mail, the formula is not very precise and the number of ULP can be a bit bigger than that. In practice the result is 0x3de829c5 instead of 0x3de829b9 for the truncated value. This means an error of 12 ULP so the error bound is not correct (because of the imprecise formula to check results).

Reply:
If I modify the checking code so to compute the error directly in ULP with the formula `abs((uniform int64)intbits(res) - (uniform int64)intbits(dres))`, then I get a maximum error of 12 ULP. This maximum error is reached for values closed to the one you mention. Note that this error is maximal for such values because of the problem mentioned above in this PR and present in the current ISPC code too (though the error is much bigger).
This means the actual error bound should not be "<10 ULP" but 12 ULP. Let say "<15 ULP" so to be safe.

Reply:
If I modify the checking code so to compute the error directly in ULP with the formula `abs((uniform int64)intbits(res) - (uniform int64)intbits(dres))`, then I get a maximum error of 12 ULP for *normal numbers*. This maximum error is reached for values closed to the one you mention. Note that this error is maximal for such values because of the problem mentioned above in this PR and present in the current ISPC code too (though the error is much bigger).

This means the actual error bound should not be "<10 ULP" but 12 ULP. Let say "<15 ULP" so to be safe.

For subnormal numbers (so values close to zeros and output close to -inf), the error is significantly bigger. I expected the mantissa to be >0.5 but I think this is not true with subnormal numbers. I need to check that.

Reply:
For the subnormal numbers, it turns out that the problem was already present in the current ISPC implementation so this is not a problem introduced by this PR. It can be fixed/mitigated, but again, at the expense of a slower implementation. Note the error can be pretty significant for small subnormal numbers. For example, for 1.20206185e-39 (0xd16dc), the results should be -89.616783 while it is -87.932335 (and -87.932327 for the current ISPC implementation which is not really better).
Tell me if you want to improve the precision for subnormal numbers (certainly at the expense of an even slower implementation).

Reply:
> 
> This means the actual error bound should not be "<10 ULP" but 12 ULP. Let say "<15 ULP" so to be safe.

This is aligned with my estimation.